src/tasks/python/PythonExtensionHelper.ts

```diff
@@ -0,0 +1,116 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// This will eventually be replaced by an API in the Python extension. See https://github.com/microsoft/vscode-python/issues/7282
+
+import * as fse from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from "vscode";
+import CliDockerClient from '../../debugging/coreclr/CliDockerClient';
+import { delay } from '../../utils/delay';
+import { PythonDefaultDebugPort, PythonFileTarget, PythonModuleTarget } from '../../utils/pythonUtils';
+
+export namespace PythonExtensionHelper {
+    export interface DebugLaunchOptions {
+        host?: string;
+        port?: number;
+        wait?: boolean;
+    }
+
+    export function getDebuggerEnvironmentVars(): { [key: string]: string } {
+        return { ["PTVSD_LOG_DIR"]: `/dbglogs` };
+    }
+
+    export function getDebuggerLogFilePath(folderName: string) : string {
+    // The debugger generates the log file with the name in this format: ptvsd-{pid}.log,
+    // So given that we run the debugger as the entry point, then the PID is guaranteed to be 1.
+        return path.join(os.tmpdir(), folderName, "ptvsd-1.log");
+    }
+
+    export async function ensureDebuggerReady(prelaunchTask: vscode.Task, debuggerSemaphorePath: string, containerName: string, cliDockerClient: CliDockerClient): Promise<void> {
+        // tslint:disable-next-line:promise-must-complete
+        return new Promise((resolve, reject) => {
+            vscode.tasks.onDidEndTask(async e => {
```

Comment:
Should unsubscribe from the event when the debugger's attached or on timeout.

Reply:
Which timeout? You mean the 10 seconds or a different one? I wanted to have a way to unsubscribe but the thing is, when should we call this a failure given the prelaunch task could be executing N number of other tasks in unknown time?

Reply:
I was referring to the 10s timeout--basically when `ensureDebuggerReady()` "completes", whether successfully or not.